x-pack/plugins/security_solution/common/api/endpoint/actions/common/base.ts

```diff
@@ -42,6 +43,13 @@ export const BaseActionRequestSchema = {
   ),
   comment: schema.maybe(schema.string()),
   parameters: schema.maybe(schema.object({})),
+  agent_type: schema.maybe(
+    schema.oneOf(
+      // @ts-expect-error TS2769: No overload matches this call
+      RESPONSE_ACTION_AGENT_TYPE.map((agentType) => schema.literal(agentType)),
+      { defaultValue: 'endpoint' }
```

Comment:
If you decide to go with an enum I would rather pass the values directly than expect ts error
schema.maybe(schema.oneOf([schema.literal(RESPONSE_ACTION_AGENT_TYPE.ENDPOINT), ...]))